CPU_consumption/Consumption/CoresConsumption.h

```diff
@@ -0,0 +1,54 @@
+//
+// Created by Nikolay Fedotenko on 03.03.2021.
+//
+
+#pragma once
+
+#include "BaseWorker.h"
+
+/*! \file
+ *  This header declares CoresConsumption class
+ */
+
+#include <mach/mach.h>
+
+/*! \class
+ *  Derived class from BaseWorker to calculate CPU consumption per cores
+ */
+class CoresConsumption : public BaseWorker
+{
+    /*! \struct
+     *  Info of a single core
+     *
+     *  \note Required to work with cores data properly via Mach API
+     */
+    typedef struct _core_info_t
+    {
+        natural_t cpu_ticks[CPU_STATE_MAX];
+        double usage;
+    } core_info_t;
```

Comment:
> Does this need to be here?

Unfortunately yes, because it's for the interaction with low-level-style interface of macOS kernel to get CPU consumption data, I'll think over a possible way to make it look more modern (probably, by implementing a kind of wrapper).